Scripts/Source/UD_RandomRestraintManager.psc

```diff
@@ -51,68 +53,85 @@ Function StartMutex()
     _mutex = true
 EndFunction
 
-
 Function EndMutex()
     _mutex = false
-    
 EndFunction
+
 bool _mutex = false
 
-;DO NOT MODIFI
+;DO NOT MODIFY
+
+;0b = zad_DeviousPiercingsVaginal
+;1b = zad_DeviousPiercingsNipple
+;2b = zad_DeviousPlugVaginal
+;3b = zad_DeviousPlugAnal
 
-;0b = zad_DeviousCollar
-;1b = zad_DeviousArmCuffs
-;2b = zad_DeviousBelt
-;3b = zad_DeviousBlindfold
+;4b = zad_DeviousHeavyBondage
+;5b = zad_deviousHarness
+;6b = zad_DeviousCorset
+;7b = zad_deviousHood
 
-;4b = zad_DeviousBra
-;5b = zad_DeviousBoots
-;6b = zad_DeviousGag
-;7b = zad_DeviousHeavyBondage
+;8b = zad_DeviousCollar
+;9b=  zad_DeviousGag
+;10b= zad_DeviousBlindfold
+;11b= zad_DeviousBoots
 
-;8b = zad_DeviousLegCuffs
-;9b= zad_DeviousSuit
-;10b= zad_DeviousPlugVaginal
-;11b= zad_DeviousPlugAnal
+;12b= zad_DeviousArmCuffs
+;13b= zad_DeviousLegCuffs
+;14b= zad_DeviousSuit
+;15b= zad_DeviousGloves
 
-;12b= zad_DeviousPiercingsVaginal
-;13b= zad_DeviousPiercingsNipple
-;14b= zad_DeviousGloves
-;15b= zad_DeviousHood
+;16b= zad_DeviousBra
+;17b= zad_DeviousBelt
 
 FormList Property UD_CheckKeywords auto
 FormList Property SuitableKeywords auto
 
 bool ready = false
+
 Event onInit()
-    UD_CheckKeywords.revert()
-    UD_CheckKeywords.addForm(libs.zad_DeviousCollar)         ;iIndex = 0
-    UD_CheckKeywords.addForm(libs.zad_DeviousArmCuffs)        ;iIndex = 1
-    UD_CheckKeywords.addForm(libs.zad_DeviousBelt)            ;iIndex = 2
-    UD_CheckKeywords.addForm(libs.zad_DeviousBlindfold)        ;iIndex = 3
-    
-    UD_CheckKeywords.addForm(libs.zad_DeviousBra)            ;iIndex = 4
-    UD_CheckKeywords.addForm(libs.zad_DeviousBoots)            ;iIndex = 5
-    UD_CheckKeywords.addForm(libs.zad_DeviousGag)            ;iIndex = 6
-    UD_CheckKeywords.addForm(libs.zad_DeviousHeavyBondage)    ;iIndex = 7
-    
-    UD_CheckKeywords.addForm(libs.zad_DeviousLegCuffs)        ;iIndex = 8
-    UD_CheckKeywords.addForm(libs.zad_DeviousSuit)            ;iIndex = 9
-    UD_CheckKeywords.addForm(libs.zad_DeviousPlugVaginal)    ;iIndex = 10
-    UD_CheckKeywords.addForm(libs.zad_DeviousPlugAnal)        ;iIndex = 11
-    
-    UD_CheckKeywords.addForm(libs.zad_DeviousPiercingsVaginal)    ;iIndex = 12
-    UD_CheckKeywords.addForm(libs.zad_DeviousPiercingsNipple)    ;iIndex = 13
-    UD_CheckKeywords.addForm(libs.zad_DeviousGloves)            ;iIndex = 14
-    UD_CheckKeywords.addForm(libs.zad_deviousHood)                 ;iIndex = 15
-    
-    
+    FillOutCheckKeywords()
     if UDmain.TraceAllowed()    
         UDCDmain.Log("-UDRRM initiated-")
     endif
     ready = True
 EndEvent
 
+Function Update()
+    FillOutCheckKeywords()
+    if UDmain.TraceAllowed()    
+        UDCDmain.Log("Refilled Keywords formlist")
+    endif
+    UD_RandomDevice_GlobalFilter = 0xFFFFFFFF
```

Comment:
This line should not exist. This will wipe all MCM settings which you added on every game reload.

Reply:
Oh... *facepalms*